### PR TITLE
[PLACES-101] Upload endpoints

### PR DESCRIPTION
**Pull request by jvisenti**
* Implements [GET /access-token/valid](https://pathcheck.atlassian.net/wiki/spaces/SA/pages/73924822/SPL+API+Specification+v1+For+MVP1#Confirmed--GET-%2Faccess-code%2Fvalid) (PLACES-112)
* Implements [POST /upload](https://pathcheck.atlassian.net/wiki/spaces/SA/pages/73924822/SPL+API+Specification+v1+For+MVP1#to-be-confirmed%3A--POST-%2Fupload) (PLACES-104)
* Implements [POST /consent](https://pathcheck.atlassian.net/wiki/spaces/SA/pages/73924822/SPL+API+Specification+v1+For+MVP1#to-be-confirmed%3A--POST-%2Fconsent) (PLACES-113)

### CODE REVIEW

File: app/api/upload/controller.js [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+const trails = require('../../../db/models/trails');
+
+/**
+ * @method checkValid
+ *
+ * Checks if a given access token is valid for use in consent and upload requests.
+ *
+ */
+exports.checkValid = async (req, res) => {
+  const { accessCode: codeId } = req.body;
```
Comment (MarshallMcCoy) [inline app/api/upload/controller.js:12]: I'm sure you're aware of this but want to make a note of it. Based on our conversation in slack we should be expecting the value of the `accessCode` here and not the `codeId`

File: db/migrations/20200528005020_access_codes.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
+exports.up = function(knex) {
+  return knex.schema.createTable('access_codes', function (table) {
+    table.string('id', 6).primary();
+    table.integer('case_id').unsigned().unique().references('cases.id').onDelete('CASCADE');
```
Comment (MarshallMcCoy) [inline db/migrations/20200528005020_access_codes.js:4]: Touched on this in an earlier comment but I think we need to include a field to persist the actual value of the access code.

File: db/migrations/20200527232927_cases.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
+const { onUpdateTrigger } = require('../../knexfile');
+
+exports.up = function(knex) {
+  return knex.schema.createTable('cases', function (table) {
+    table.integer('id').unsigned().primary();
+    table.boolean('consent');
```
[the remark this answers is not in the document]
Comment (jvisenti) [reply db/migrations/20200527232927_cases.js:6]: 👍

File: app/api/upload/controller.js (hunk 1 above)
Comment (jvisenti) [reply app/api/upload/controller.js:12]: Not to confuse this PR with the conversation in Slack, but here I implemented the "value" of the access token as the primary key and named it `id`.

File: db/migrations/20200528005020_access_codes.js (hunk 2 above)
Comment (jvisenti) [reply db/migrations/20200528005020_access_codes.js:4]: `id` is the 6-digit value and primary key, possibly confusing given the conversation happening in Slack

**Comment by jvisenti** [comment]
Updated to match [latest spec](https://pathcheck.atlassian.net/wiki/spaces/SA/pages/96076138/v2+Upload+Flow)

**Pull request merged**